zebra-network/src/address_book.rs

```diff
@@ -159,6 +159,22 @@ impl AddressBook {
         let mut peers = peers
             .values()
             .filter_map(MetaAddr::sanitize)
+            .filter(|address| {
+                // Security: remove peers that:
+                //   - last responded more than three hours ago, or
+                //   - haven't responded yet but were reported last seen more than three hours ago
+                address
+                    .last_seen()
+                    // Correctness: the last seen shouldn't ever be in the future, either because
+                    // we set the time or because another peer's future time was sanitized when it
+                    // was added to the address book
+                    .map(|last_seen| {
+                        last_seen.saturating_elapsed() < constants::REACHABLE_PEER_DURATION
+                    })
+                    // Correctness: remove peers that haven't been seen or reported to have been
+                    // seen
+                    .unwrap_or(false)
```

Comment:
Testing: this check might be easier to test as a MetaAddr method.

Code structure: using a MetaAddr method would also remove the 3-layer nested code.

Reply:
Done in [189f0d1](https://github.com/jvff/zebra/commit/189f0d178fccb104c8bd0776cac6b2db7bc84f4d). Much easier to read now, thanks!